lib/ui/components/styled_components/rich_text.widget.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:flutter/material.dart';
+import 'package:yc_app_utils/helpers/helpers.dart';
+import 'package:yc_app_utils/models/styled_component/rich_text.model.dart';
+
+class RichTextWidget extends StatefulWidget {
+  final RichTextModel richText;
+  const RichTextWidget({
+    required this.richText,
+    super.key,
+  });
+
+  @override
+  State<RichTextWidget> createState() => _RichTextWidgetState();
+}
+
+class _RichTextWidgetState extends State<RichTextWidget> {
+  @override
+  Widget build(BuildContext context) {
+    return widget.richText.texts?.isEmpty ?? true
+        ? const SizedBox.shrink()
+        : Container(
+            padding: CommonHelpers.getPaddingFromList(
+              widget.richText.primaryTextStyle?.padding,
+            ),
+            decoration: CommonHelpers.getBoxDecorationWithSectionBackground(
+                sectionBackground:
+                    widget.richText.primaryTextStyle?.background),
+            child: RichText(
+              textAlign: widget.richText.primaryTextStyle?.textAlign ??
+                  TextAlign.start,
+              maxLines: widget.richText.primaryTextStyle?.maxLines,
+              overflow: widget.richText.primaryTextStyle?.textOverflow ??
+                  TextOverflow.clip,
+              text: TextSpan(
+                style: CommonHelpers.getTextStyleFromV2TextStyle(
+                  widget.richText.primaryTextStyle,
+                ),
+                children: List.generate(
+                  widget.richText.texts?.length ?? 0,
```

Comment:
```suggestion
                  widget.richText.texts!.length,
```
as already verified at the start